### DISCUSSION TITLE
Does "Black Magic Debug" support the software development of the upper computer, directly read the address information of the peripheral MCU (ARM core) through the swd interface

### DISCUSSION THREAD

**Discussion by qwer4650987** [blackmagic-debug/blackmagic#1839] [Q&A]
I am the first contact with "Black Magic Debug", my need is to develop a host computer software, directly through the swd interface to read the peripheral MCU (ARM core) address information, to achieve the burning function, my question: 
1. Can this software meet my needs? 
2. If it satisfies the requirements, how can it be compiled into a DLL dynamic library for C++ calling? 
3. Does this "Black Magic Debug" method directly support J-link or st-link emulators? Can that directly read the Flash address information through the emulator RTT mode?

**Comment by dragonmux** [comment]
If we understand your question, you're asking if you could write BMDA - which already does exactly what you're after.

If BMDA doesn't satisfy your needs, technically there is libbmd under the Meson build system. This gives one access to the core project (all the scan implementations, the GDB server, target support, and architecture support) as a shared library/DLL for embedding and linking - however, it does not give you access to any of BMDA's functionality to talk to ST-Links, J-Links, CMSIS-DAP adaptors, etc.

See [the BMDA README.md](https://github.com/blackmagic-debug/blackmagic/tree/main/src/platforms/hosted) for details about that part of the project.
